src/goto-programs/goto_convert.cpp

```diff
@@ -1838,6 +1842,28 @@ void goto_convertt::generate_ifthenelse(
     false_case.instructions.back().guard=guard;
     dest.destructive_append(false_case);
     false_case.instructions.clear();
+    if(
+      is_empty(true_case) ||
+      (is_size_one(true_case) &&
+       is_skip(true_case, true_case.instructions.begin())))
+      return;
+  }
+
+  // a special case for C libraries that use
+  // (void)((cond) || (assert(0),0))
+  if(
+    is_empty(false_case) && true_case.instructions.size() == 2 &&
```

Comment:
Is there any reason to restrict ourselves to two instructions here? Could we simply say "if(x) { assert(false); ... } --> assert(!x)", since the rest of the true-case is dead code? Also, why only consider the true case?

Reply:
It's only dead code if none of the instruction has a label that could be jumped to.